### DISCUSSION TITLE
Send client_id + client_secret in Authorization to TokenEndpoint

### DISCUSSION THREAD

**Discussion by JJRdec** [soofstad/react-oauth2-pkce#117] [Q&A] [answered]
### Summary

Our Authorization Server expects a Basic Auth header included when requests are made to TokenEndpoint. 

Please could you make this an option in the config, many thanks!

### Basic Example

Authorization=base64Encoded(client_id, cliend_secret)

```
POST /oauth2/token?client_id=client&
  redirect_uri=https://client.example.com/authorized&
  grant_type=authorization_code&
  code=Gl_8xcHRQumJSLO5HQd9xfgY8...
  Host: as.example.com
  Authorization: Basic czZCaGRSa3F0MzpnWDFmQmF0M2JW
```

### Drawbacks

Without this feature, the lib will be incompatible with some Authorization Servers

**Comment by anthony-vito** [comment, marked as answer]
This library only supports [RFC-7636](https://www.rfc-editor.org/info/rfc7636) and is not applicable to the authorization code flows that OAuth2 lays out in [RFC-6749](https://www.rfc-editor.org/info/rfc6749). ( using client secrets ) that you are requesting.

The main concept at work here is that this is a "front end only" library that interacts directly with an authorization server to perform an authorization flow and get an access token. Being a front end only library there is no way to keep the client secret, well, secret! To meet this need, an extension to OAuth2, RFC-7636 "Proof Key for Code Exchange by OAuth Public Clients" was created. This library implements that specification.

If you want to use authorization code flows using client secrets you need to make sure you are using a backend server that holds your client secret and coordinates the authorization. There are plenty of backend frameworks written in Javascript using Node that accomplish this if that is your goal.
